core/src/main/java/io/questdb/std/str/StringSink.java

```diff
@@ -82,12 +82,20 @@ public int hashCode() {
         return Chars.hashCode(buffer, 0, pos);
     }
 
-    public int indexOf(String s) {
+    public int indexOf(@NotNull String s) {
         return Chars.indexOf(this, 0, pos, s);
     }
 
-    public int indexOf(String s, int fromIndex) {
-        return Chars.indexOf(this, fromIndex, pos, s);
+    public int indexOf(@NotNull String s, int fromIndex) {
+        return Chars.indexOf(this, Math.min(fromIndex, pos), pos, s);
+    }
+
+    public int lastIndexOf(@NotNull String s) {
```

Comment:
this looks unused. apart from tests.

Reply:
Yes, it's currently unused, but I added it to keep the API complete and consistent.